src/main/java/com/linkedin/kmf/services/SignalFxMetricsReporterService.java

```diff
@@ -0,0 +1,197 @@
+  }
+
+  @Override
+  public synchronized void start() {
+    _executor.scheduleAtFixedRate(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          captureMetrics();          
+        } catch (Exception e) {
+          LOG.error(_name + "/SignalFxMetricsReporterService failed to report metrics", e);
+        }
+      }
+    }, _reportIntervalSec, _reportIntervalSec, TimeUnit.SECONDS);
+    LOG.info("{}/SignalFxMetricsReporterService started", _name);
+  }
+
+  @Override
+  public synchronized void stop() {
+    _executor.shutdown();
+    LOG.info("{}/SignalFxMetricsReporterService stopped", _name);
+  }
+
+  @Override
+  public boolean isRunning() {
+    return !_executor.isShutdown();
+  }
+
+  @Override
+  public void awaitShutdown() {
+    try {
+      _executor.awaitTermination(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
+    } catch (InterruptedException e) {
+      LOG.info("Thread interrupted when waiting for {}/SignalFxMetricsReporterService to shutdown", _name);
+    }
+    LOG.info("{}/SignalFxMetricsReporterService shutdown completed", _name);
+  }
+  
+  private void setUpDimensionMap() {
+    if (_dimensions.size() == 0) {
+      return;
+    }
+    for (String dimension : _dimensions) {
+      String dimensionKey = dimension.split(":")[0];
+      String dimensionValue = dimension.split(":")[1];
+      _dimensionsMap.put(dimensionKey, dimensionValue);
+    }
+  }
+  
+  private SignalFxEndpoint getSignalFxEndpoint(String urlStr) throws Exception {    
+    URL url = new URL(urlStr);
+    String scheme = url.getProtocol();
+    String host = url.getHost();
+    int port = url.getPort();
+    return new SignalFxEndpoint(scheme, host, port);    
+  }
+
+  private String generateSignalFxMetricName(String bean, String attribute) {    
+    String service = bean.split(":")[1];
+    String serviceType = service.split(",")[1].split("=")[1];
+    String[] segs = {serviceType, attribute};
+    return StringUtils.join(segs, ".");
+  }
+
+  private void captureMetrics() {
+    for (String metricName : _metricNames) {
+      String mbeanExpr = metricName.substring(0, metricName.lastIndexOf(":"));
+      String attributeExpr = metricName.substring(metricName.lastIndexOf(":") + 1);
+      
+      List<MbeanAttributeValue> attributeValues = getMBeanAttributeValues(mbeanExpr, attributeExpr);
+
+      for (final MbeanAttributeValue attributeValue : attributeValues) {
+        String metric = attributeValue.toString();
+        String key = metric.substring(0, metric.lastIndexOf("="));
+        String[] parts = key.split(",");
+        if (parts.length < 2) {
+          continue;
+        }
+        parts = parts[0].split("=");
+        if (parts.length < 2 || !parts[1].contains("cluster-monitor")) {
+          continue;
+        }
+        String signalFxMetricName = generateSignalFxMetricName(attributeValue.mbean(), attributeValue.attribute());        
```

Comment:
It's a bit sad that you have to recompute this every time when the set of metrics doesn't change. Can you maintain a map of `MbeanAttributeValue` to `SettableDoubleGauge`?